includes/tasks/class-wc-calypso-task-add-domain.php

```diff
@@ -67,8 +67,19 @@ public function get_time() {
 	public function get_action_url() {
 		$status      = new \Automattic\Jetpack\Status();
 		$site_suffix = $status->get_site_suffix();
+		$domain_path = sprintf( "https://wordpress.com/domains/add/%s", $site_suffix );
+		$home_url    = \home_url( '', 'https' );
 
-		return sprintf( "https://wordpress.com/domains/add/%s", $site_suffix );
+		if ( ! \str_starts_with( $home_url, 'https://woo-' ) && ! \str_starts_with( $home_url, 'https://wooexpress-' ) ) {
```

Comment:
Hey @daledupreez `str_starts_with` is a PHP8 function - is there any chance that we end up with a PHP7.x installation?

I don't think it needs a `\` when calling it, as it's a built-in function that exists in the global namespace. 

Same for `home_url` and `get_option` - they do exist in the global namespace

Reply:
> `str_starts_with` is a PHP8 function - is there any chance that we end up with a PHP7.x installation?

Good catch, @PanosSynetos! However, WordPress Core has had a polyfill available since 5.9.0 (at least per [this code](https://github.com/WordPress/wordpress-develop/blob/7c965e781f297e9f47f3abd3835c613eb41c0765/src/wp-includes/compat.php#L467-L489)). And we are running on at least WordPress 6.1 on Atomic servers.

> I don't think it needs a `\` when calling it, as it's a built-in function that exists in the global namespace.
> Same for `home_url` and `get_option` - they do exist in the global namespace

With the file namespaced, I wanted to be explicit about what I want to happen. So while these aren't _necessary_, it's very clear what they ought to be doing.